packages/udt/src/udt/index.ts

```diff
@@ -274,33 +367,514 @@ export class Udt extends ssri.Trait {
     return resTx;
   }
 
-  async completeChangeToLock(
+  /**
+   * Checks if a cell is a valid UDT cell for this token.
+   * A valid UDT cell must have this UDT's type script and contain at least 16 bytes of output data
+   * (the minimum required for storing the UDT balance as a 128-bit little-endian integer).
+   *
+   * @param cellOutputLike - The cell output to check
+   * @param outputData - The output data of the cell
+   * @returns True if the cell is a valid UDT cell for this token, false otherwise
+   *
+   * @example
+   * ```typescript
+   * const udt = new Udt(codeOutPoint, scriptConfig);
+   * const cellOutput = { lock: someLock, type: udt.script };
+   * const outputData = ccc.numLeToBytes(1000, 16); // 1000 UDT balance
+   *
+   * const isValid = udt.isUdtCell(cellOutput, outputData);
+   * console.log(`Is valid UDT cell: ${isValid}`); // true
+   * ```
+   *
+   * @remarks
+   * The method checks two conditions:
+   * 1. The cell's type script matches this UDT's script
+   * 2. The output data is at least 16 bytes long (required for UDT balance storage)
+   */
+  isUdtCell(cellOutputLike: ccc.CellOutputLike, outputData: ccc.HexLike) {
+    return (
+      (ccc.CellOutput.from(cellOutputLike).type?.eq(this.script) ?? false) &&
+      ccc.bytesFrom(outputData).length >= 16
+    );
+  }
+
+  /**
+   * Calculates the total UDT balance from all inputs in a transaction.
+   * This method examines each input cell and sums up the UDT amounts
+   * for cells that have this UDT's type script.
+   *
+   * @param txLike - The transaction to analyze
+   * @param client - The client to fetch input cell data
+   * @returns A promise resolving to the total UDT balance from all inputs
+   *
+   * @example
+   * ```typescript
+   * const udt = new Udt(codeOutPoint, scriptConfig);
+   * const tx = ccc.Transaction.from(existingTransaction);
+   *
+   * const inputBalance = await udt.getInputsBalance(tx, client);
+   * console.log(`Total UDT input balance: ${inputBalance}`);
+   * ```
+   *
+   * @remarks
+   * This method only counts inputs that have the same type script as this UDT instance.
+   * Inputs without a type script or with different type scripts are ignored.
+   */
+  async getInputsBalance(
```

Comment:
Why not also returning the CKB occupied by said UDT cells?

Reply:
Good idea!